internal/driver/allocation_settings_test.go

```diff
@@ -0,0 +1,200 @@
+package driver
+
+import (
+	"testing"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func Test_MinimumAllocationSettings(t *testing.T) {
+	mockBlock := &csi.VolumeCapability{AccessType: &csi.VolumeCapability_Block{
+		Block: &csi.VolumeCapability_BlockVolume{},
+	}}
+	mockMount := &csi.VolumeCapability{AccessType: &csi.VolumeCapability_Mount{
+		Mount: &csi.VolumeCapability_MountVolume{FsType: "ext4"},
+	}}
+
+	type testBytes struct {
+		required, limit int64
+	}
+	testCases := []struct {
+		name string
+
+		settings ControllerAllocationSettings
+
+		capabilities []*csi.VolumeCapability
+
+		input    testBytes
+		expected testBytes
+	}{
+		{
+			name:     "no settings result in pass through",
+			settings: ControllerAllocationSettings{},
+			input: testBytes{
+				required: 1 << 30,
+				limit:    2 << 30,
+			},
+			expected: testBytes{
+				required: 1 << 30,
+				limit:    2 << 30,
+			},
+		},
+		{
+			name: "default minimum should be applied for block storage",
```

Comment:
In this case, the minimum value is not the default value, but 1Gi. So the name should be "minimum should be applied for block storage".